Dockerfile

```diff
@@ -92,134 +109,106 @@ ARG VERSION_PKG="github.com/talos-systems/talos/internal/pkg/version"
-
-# The binaries target allows for parallel compilation of all binaries.
-
-FROM scratch AS binaries-build
-COPY --from=init / /
-COPY --from=osd / /
-COPY --from=trustd / /
-COPY --from=proxyd / /
-COPY --from=ntpd / /
-COPY --from=osctl-linux-amd64 / /
-COPY --from=osctl-darwin-amd64 / /
-
-FROM scratch AS binaries
-COPY --from=binaries-build /osctl-linux-amd64 /osctl-linux-amd64
-COPY --from=binaries-build /osctl-darwin-amd64 /osctl-darwin-amd64
+WORKDIR /src/cmd/osctl
+RUN GOOS=darwin GOARCH=amd64 go build -a -ldflags "-s -w -X ${VERSION_PKG}.Name=Client -X ${VERSION_PKG}.SHA=${SHA} -X ${VERSION_PKG}.Tag=${TAG}" -o /osctl-darwin-amd64
+RUN chmod +x /osctl-darwin-amd64
+FROM scratch AS osctl-darwin
+COPY --from=osctl-darwin-build /osctl-darwin-amd64 /osctl-darwin-amd64
 
 # The kernel target is the linux kernel.
 
-ARG KERNEL_IMAGE
-FROM ${KERNEL_IMAGE} as kernel
+FROM scratch AS kernel
+COPY --from=docker.io/autonomy/kernel:2ac99a0 /boot/vmlinuz /vmlinuz
 
-# The initramfs target creates the compressed initramfs.
+# The initramfs target provides the Talos initramfs image.
 
-FROM base AS init-build
-ARG SHA
-ARG TAG
-ARG VERSION_PKG="github.com/talos-systems/talos/internal/pkg/version"
-WORKDIR /src/internal/app/init
-RUN --mount=type=cache,target=/root/.cache go build -a -ldflags "-s -w -X ${VERSION_PKG}.Name=Talos -X ${VERSION_PKG}.SHA=${SHA} -X ${VERSION_PKG}.Tag=${TAG}" -o /init
-RUN chmod +x /init
+FROM tools AS initramfs-build
+COPY --from=docker.io/autonomy/fhs:8467184 / /rootfs
+COPY --from=docker.io/autonomy/ca-certificates:20f39f7 / /rootfs
+COPY --from=docker.io/autonomy/dosfstools:767dee6 / /rootfs
+COPY --from=docker.io/autonomy/musl:9bc7430 / /rootfs
+COPY --from=docker.io/autonomy/syslinux:85e1f9c / /rootfs
+COPY --from=docker.io/autonomy/xfsprogs:5e50579 / /rootfs
+COPY ./hack/cleanup.sh /toolchain/bin/cleanup.sh
+RUN ["/toolchain/bin/cleanup.sh", "/rootfs"]
 
-FROM scratch AS init
-COPY --from=init-build /init /init
+FROM scratch AS initramfs-base
+COPY --from=initramfs-build /rootfs /
+COPY --from=init /init /init
 
-ARG INITRAMFS_IMAGE
-FROM ${INITRAMFS_IMAGE} AS initramfs-build
-WORKDIR /
-COPY --from=init-build /init /init
-
-ARG TOOLCHAIN_IMAGE
-FROM ${TOOLCHAIN_IMAGE} AS initramfs-archive
-COPY --from=initramfs-build / /initramfs
+FROM build AS initramfs-archive
+COPY --from=initramfs-base / /initramfs
 WORKDIR /initramfs
 RUN set -o pipefail && find . 2>/dev/null | cpio -H newc -o | xz -v -C crc32 -0 -e -T 0 -z >/initramfs.xz
 
 FROM scratch AS initramfs
 COPY --from=initramfs-archive /initramfs.xz /initramfs.xz
 
-# The rootfs target creates the root filesystem archive.
-
-ARG ROOTFS_IMAGE
-FROM ${ROOTFS_IMAGE} AS rootfs-build
-COPY --from=kernel /modules /
-COPY images /usr/images
-
-ARG TOOLCHAIN_IMAGE
-FROM ${TOOLCHAIN_IMAGE} AS rootfs-archive
-COPY --from=rootfs-build / /rootfs
+# The rootfs target provides the Talos rootfs image.
+
+FROM tools AS rootfs-build
+COPY --from=docker.io/autonomy/fhs:8467184 / /rootfs
```

Comment:
it probably might make sense in the future to use some more human-readable tag names? so that it's easier to track when deps are going to be updated. certainly not for this PR, but as a future todo

Reply:
Yeah I'm hoping that we can use SHAs while still in alpha and then cut official releases of the images. There is also probably some more optimization we can do around setting these dynamically using a dependency algorithm in `bldr`. I have some ideas here. Actually I was hoping to chat with you about them.